src/include/type/type_util.h

```diff
@@ -92,6 +92,46 @@ class TypeUtil {
       }
     }
   }
+
+  static type::TypeId TypeIdFromString(const std::string &type_string) {
+    if (type_string == "INVALID") {
```

Comment:
This should be a switch.

Reply:
I don't think c++ supports switch on strings?

Reply:
Right, forget you were looking at strings...